src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -199,44 +271,67 @@ public void PacketReset(SNIHandle handle, bool write, SNIPacket packet)
         /// <param name="async">Asynchronous connection</param>
         /// <param name="parallel">Attempt parallel connects</param>
         /// <returns>SNI handle</returns>
-        public SNIHandle CreateConnectionHandle(object callbackObject, string fullServerName, bool ignoreSniOpenTimeout, long timerExpire, out byte[] instanceName, byte[] spnBuffer, bool flushCache, bool async, bool parallel)
+        public SNIHandle CreateConnectionHandle(object callbackObject, string fullServerName, bool ignoreSniOpenTimeout, long timerExpire, out byte[] instanceName, ref byte[] spnBuffer, bool flushCache, bool async, bool parallel)
         {
             instanceName = new byte[1];
             instanceName[0] = 0;
 
             string[] serverNameParts = fullServerName.Split(':');
 
-            if (serverNameParts.Length > 2)
+            // when no protocol specified
+            // serverNameParts.Length == 1 -> fullServerName is in hostname or ipv4 format
+            // serverNameParts.Length == 8 -> fullServerName is in ipv6 format
+            if (serverNameParts.Length == 1 || serverNameParts.Length == 8)
             {
-                SNILoadHandle.SingletonInstance.LastError = new SNIError(SNIProviders.INVALID_PROV, 0, SNICommon.InvalidConnStringError, string.Empty);
-                return null;
+                // Default to using tcp if no protocol is provided
+                return CreateTcpHandle(fullServerName, timerExpire, callbackObject, parallel, ref spnBuffer);
             }
-
-            // Default to using tcp if no protocol is provided
-            if (serverNameParts.Length == 1)
+            // when protocol specified
+            // serverNameParts.Length == 2 -> <protocol>:<hostname_or_ipv4> format
+            // serverNameParts.Length == 9 -> <protocol>:<hostname_or_ipv6> format
+            else if (serverNameParts.Length == 2 || serverNameParts.Length == 9)
             {
-                return CreateTcpHandle(serverNameParts[0], timerExpire, callbackObject, parallel);
-            }
+                int protocolLength = serverNameParts[0].Length + 1; // including ':'
+                string serverNameWithOutProtocol = fullServerName.Substring(protocolLength, fullServerName.Length - protocolLength);
 
-            switch (serverNameParts[0])
+                switch (serverNameParts[0])
+                {
+                    case TdsEnums.TCP:
+                        return CreateTcpHandle(serverNameWithOutProtocol, timerExpire, callbackObject, parallel, ref spnBuffer);
+
+                    case TdsEnums.NP:
+                        return CreateNpHandle(serverNameWithOutProtocol, timerExpire, callbackObject, parallel);
```

Comment:
What happens to Integrated Auth when we use Named Pipes ? This may not be a valid scenario for Unix as NP is supported only locally and the Linux Sql Server doesn't support NP yet, but we need to light up the feature on Windows as well. 
The spnBuffer is not being sent to NamedPipe code path.